subsys/bluetooth/controller/multithreading_lock.c

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include "multithreading_lock.h"
+
+/* Semaphore with a single count functioning as a lock. */
+K_SEM_DEFINE(ble_controller_lock, 1, 1);
+
+int32_t multithreading_lock_acquire(int32_t timeout)
+{
+#if IS_ENABLED(CONFIG_BLECTLR_THREADSAFETY)
+	return k_sem_take(&ble_controller_lock, K_MSEC(timeout));
```

Comment:
We should treat the K_NO_WAIT and K_FOREVER in some special way. We can argue about K_NO_WAIT - as it is 0 and in most cases even in K_MSEC makes any operation would return in 0, but K_FOREVER, being (-1) can be changed by the operation of K_MSEC macro in future implementation. I mean... now it works, but assuming that K_MSEC does no operation at all may lead us just to removing it.

Or we can assume that if K_MSEC macro performs any operation it would treat K_FOREVER in special manner - but again - then we need some documentation there.